packages/utils/telemetry-utils/src/test/config.spec.ts

```diff
@@ -122,6 +123,27 @@ describe("Config", () => {
         assert.equal(config.getBooleanArray("badBooleanArray2"), undefined);
     });
 
+    it("Typing - custom provider - With version", () => {
+        const settings = {};
+        settings[`number-${pkgVersion}`] = 1;
+        settings[`string-${pkgVersion}`] = "string";
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        settings[`string`] = "string1";
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        settings["boolean"] = true;
+
+        const mockStore = untypedProvider(settings);
+        const config = new CachedConfigProvider(mockStore);
+
+        assert.equal(config.getNumber("number"), 1);
+
+        assert.equal(config.getString("string"), "string");
+        assert.equal(config.getString("badString"), undefined);
```

Comment:
not sure of the intent for the `badString` and `badBoolean` here, as they're not declared in the original settings object as actually bad values - which can't be parsed. For example, in the other tests they're like `badString: []` and ` badBoolean: "truthy",`


Same for the other test.